src/main/java/org/assertj/core/api/recursive/comparison/RecursiveComparisonDifferenceCalculator.java

```diff
@@ -126,9 +127,10 @@ private void initDualValuesToCompare(Object actual, Object expected, FieldLocati
           if (expectedFieldsNames.containsAll(nonIgnoredActualFieldsNames)) {
             // we compare actual fields vs expected, ignoring expected additional fields
             for (String nonIgnoredActualFieldName : nonIgnoredActualFieldsNames) {
+              IntrospectionStrategy introspectionStrategy = recursiveComparisonConfiguration.getIntrospectionStrategy();
               DualValue fieldDualValue = new DualValue(fieldLocation.field(nonIgnoredActualFieldName),
-                                                       COMPARISON.getSimpleValue(nonIgnoredActualFieldName, actual),
-                                                       COMPARISON.getSimpleValue(nonIgnoredActualFieldName, expected));
+                                                       introspectionStrategy.getValue(nonIgnoredActualFieldName, actual),
+                                                       introspectionStrategy.getValue(nonIgnoredActualFieldName, expected));
```

Comment:
That's a good first step, we need to go further, `RecursiveComparisonDifferenceCalculator` should not use at all `COMPARISON` nor `Objects.getDeclaredFieldsIncludingInherited` or `Objects.getFieldsNames` since these are ways to get members to use in the comparison (indirectly sometimes).

`RecursiveComparisonDifferenceCalculator` is a complex piece of code by itself, it should delegate member introspection to `IntrospectionStrategy`, that will move a bit of complexity out of it and make it clear when introspection is used.

Reply:
I see. Maybe new type of Assert should be introduced to existing methods naming `ignoringFieldsMatchingRegexes` etc. if we are bringing the `member` terminology. As for now I will try to work your comment and see how it turns out.

Reply:
I need to give some thoughts about it, I'm not sure I want to deprecate a good part of the API mentioning fields and introducing `ignoringMembersMatchingRegexes` might not be that clear ... let me think about it.

Reply:
did not have the time this week, hopefully in the week end but no promises

Reply:
Sure, I am doing some slow and steady research 😄